### PR TITLE
Add note about backtraces to TRPL

### PR DESCRIPTION
**Pull request by steveklabnik**
This is where panic!() is introduced.

### CODE REVIEW

File: src/doc/trpl/functions.md [hunk 1]
```diff
@@ -214,11 +214,61 @@ fn diverges() -> ! {
...
+  10:     0x7f4027738a19 - main
+  11:     0x7f402694ab44 - __libc_start_main
+  12:     0x7f40277386c8 - <unknown>
+  13:                0x0 - <unknown>
+```
+
+`RUST_BACKTRACE` also works with Cargo’s `run` command:
+
+```text
+$ cargo run
```
Comment (alexcrichton) [inline src/doc/trpl/functions.md]: Should this have RUST_BACKTRACE in front of it?

File: src/doc/trpl/functions.md [hunk 2]
```diff
@@ -214,11 +214,61 @@ fn diverges() -> ! {
...
+some output that looks like this:
+
+```text
+thread ‘<main>’ panicked at ‘This function never returns!’, hello.rs:2
+```
+
+If you want more information, you can get a backtrace by setting the
+`RUST_BACKTRACE` environment variable:
+
+```
```
Comment (alexcrichton) [inline src/doc/trpl/functions.md]: This'll need to be tagged `text`, right?

File: src/doc/trpl/functions.md (hunk 1 above)
Comment (steveklabnik) [inline src/doc/trpl/functions.md]: whoops!

**Comment by steveklabnik** [comment]
@alexcrichton totally right. addressed both nits

**Pull request merged**